dockerbuild/ssp-overrides/sp-php.patch

```diff
@@ -1,16 +1,39 @@
-811a812,819
->         /*
->          * If this SP is allowed to use more than one IdP, then send to discovery page
->          */
->         if (sizeof($state['saml:IDPList']) > 1) {
->             $state['LoginCompletedHandler'] = [SP::class, 'reauthPostLogin'];
->             $this->authenticate($state);
->         }
-> 
-861,863c869,870
-<             $state['saml:sp:IdPMetadata'] = $this->getIdPMetadata($state['saml:sp:IdP']);
-<             $state['saml:sp:AuthId'] = $this->authId;
-<             self::askForIdPChange($state);
----
->             $state['LoginCompletedHandler'] = [SP::class, 'reauthPostLogin'];
->             $this->authenticate($state);
+*** ../../vendor/simplesamlphp/simplesamlphp/modules/saml/src/Auth/Source/SP.php	2024-04-30 10:38:57.000000000 -0600
+--- SP.php	2024-07-11 17:07:40.000000000 -0600
+***************
+*** 809,814 ****
+--- 809,822 ----
+              $state[$k] = $v;
+          }
+  
++         /*
++          * If this SP is allowed to use more than one IdP, then send to discovery page
++          */
++         if (sizeof($state['saml:IDPList']) > 1) {
++             $state['LoginCompletedHandler'] = [SP::class, 'reauthPostLogin'];
++             $this->authenticate($state);
++         }
++ 
+          // check if we have an IDPList specified in the request
+          if (
+              isset($state['saml:IDPList'])
+***************
+*** 858,866 ****
+                  $state['core:SP']
+              ));
+  
+!             $state['saml:sp:IdPMetadata'] = $this->getIdPMetadata($state['saml:sp:IdP']);
+!             $state['saml:sp:AuthId'] = $this->authId;
+!             self::askForIdPChange($state);
```

Comment:
Does the `!` indicate a conflict of some sort? I see it used to be `<` and `>`, but now it's `!`.

Reply:
No, it just means "changed line". Not sure if my commit message was clear, but from what I found today, this patch format is recommended because it includes context. Before this change, I noticed that the first "chunk" could be inserted anywhere in the file, which seemed dangerous to me.

Reply:
Your commit message does make sense, thanks. I'm still a little confused why the patch-with-context stopped showing before-and-after for that second chunk, and now merely shows the result of the change. That's all, thanks.

Reply:
No problem. I'm still learning the details of this myself. The "before" is still there. It's the portion after the `*** 858,866 ****` line, whereas the "after" lines follow the `--- 866,873 ----` line.

Reply:
Ahh, okay. Thanks.